### PR TITLE
Fix/plat 1376 exception handler

### PR DESCRIPTION
**Pull request by andela-sjames**
PR: https://themuse.atlassian.net/browse/PLAT-1376

Fixes Exception handler logging in Dev

https://app.datadoghq.com/logs?cols=%5B%22core_host%22%2C%22core_service%22%5D&event&from_ts=1548608753478&index=main&live=true&query=service%3Aweb+env%3Adev&stream_sort=desc&to_ts=1548695153478

NB: The latest logs reflects the traceback, type and value handled properly.

### CODE REVIEW

File: muselog/datadog.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import inspect
```
Comment (Shikkic) [inline muselog/datadog.py]: Is this being used?

**Comment by Shikkic** [comment]
@andela-sjames Where did you get the idea for this, was it from the Datadog logs? Can you maybe explain what's going on here and why the changes were necessary?

Also do we still want to update the usage of `app_log.logger` references in web, since that isn't leveraging muselogger? Or is that not a problem?

**Comment by andela-sjames** [comment]
> @andela-sjames Where did you get the idea for this, was it from the Datadog logs? Can you maybe explain what's going on here and why the changes were necessary?
> 
> Also do we still want to update the usage of `app_log.logger` references in web, since that isn't leveraging muselogger? Or is that not a problem?


@Shikkic 
After going through the docs (Tornado docs), I realised there are two more logs provided by tornado. 

Logger objects for internal tornado use
```
access_log = logging.getLogger("tornado.access")
app_log = logging.getLogger("tornado.application")
gen_log = logging.getLogger("tornado.general")
```

and based on this definition
```
* ``tornado.access``: Per-request logging for Tornado's HTTP servers (and
  potentially other servers in the future)
* ``tornado.application``: Logging of errors from application code (i.e.
  uncaught exceptions from callbacks)
* ``tornado.general``: General-purpose logging, including any errors
  or warnings from Tornado itself.
```
I figured what we had there was okay, then realised that unlike graylog we had to do the formatting ourself, that made me go back to how we handle traceback from muselog did a back and forth testing and figures a way to fine-tune the exception when it is converted to Json.

File: muselog/datadog.py [hunk 2]
```diff
@@ -60,16 +60,41 @@ def makePickle(self, record: LogRecord) -> str:
...
+class ObjectEncoder(json.JSONEncoder):
+    def default(self, obj):
+        if hasattr(obj, "to_json"):
+            return self.default(obj.to_json())
+        elif hasattr(obj, "__dict__"):
+            return obj.__class__.__name__
+        elif hasattr(obj, "tb_frame"):
+            return "traceback"
+
+        return super(ObjectEncoder, self).default(obj)
```
Comment (ian-axelrod) [inline muselog/datadog.py]: This should just be `super().default(obj)`

**Comment by andela-sjames** [comment]
Updated @Shikkic @ian-axelrod

**Pull request merged**